posts/cellular-neighborhoods/cellular-neighborhoods.qmd

```diff
@@ -0,0 +1,270 @@
+
+Cellular neighborhood analysis begins with two steps:
+
+1. Define each cell's neighboring cells. 
+2. Create a new matrix encoding cells' "spatial contexts". 
+To do this, we compute summaries of each cell's neighbors, reporting for example
+ their average expression profile or the abundance of different cell types within them.
+
+```{r, out.width="120%"}
+#| eval: true
+#| echo: false
+#| fig-cap: "Left: example of a cellular neighborhood defined as a cell's 50 nearest neighbors. Right: the mean expression profile of those neighbors is used to define the cell's spatial context. This 'mean neighborhood expression', calculated separately for all cells in the dataset, defines a new data matrix." 
+knitr::include_graphics("./figures/cartoon.png")
+```
+
+Once we've obtained a matrix of spatial context data, myriad analyses become possible. 
+We can:
+
+- Plot spatially smoothed expression to make spatial expression patterns for visually clear
+- Derive spatial clusters / niches
+- Find spatially auto-correlated genes / sets of genes
+- Set up interesting differential expression problems, asking how cells modulate expression in response to their spatial context
+- Explore ligand-receptor interactions
+
+Many of these analyses are achieved by simply applying techniques from single cell analyses,
+for example clustering the rows (cells) or columns (genes) of the matrix. 
+
+
+# Tools for analyzing cellular neighborhoods
+
+A small R package implementing the building blocks of cellular neighborhood analysis is 
+[here](/_code/cellular-neighborhoods){target="_blank"}. 
+
+To install it:
+
+```{r, echo = TRUE, eval = FALSE}
+devtools::install_github("Nanostring-Biostats/CosMx-Analysis-Scratch-Space@Main", subdir = "_code/cellular-neighborhoods")
+```
+
+To set up our examples, let's load the package and open its data:
+
+```{r, echo = TRUE, eval = TRUE}
+library(CellularNeighborhoods)
+data(cosmx_kidney)
+annot <- cosmx_kidney$annot
+rownames(annot) <- annot$cell_ID
+counts <- cosmx_kidney$counts
+celltype <- as.factor(cosmx_kidney$annot$celltype)
+xy <- cosmx_kidney$xy
+```
+
+Now we're ready to demonstrate the basics.
+
+## Defining a cell's neighbors
+ 
+Convenient approaches to define a cell's neighbors include the "K-nearest" approach
+(we usually use the nearest 50 neighbors) and a radius-based approach. 
+We prefer the K-nearest neighbors approach, mainly because radius-based neighborhoods
+ tend to vary widely in the number of cells they contain, and neighborhoods of very few cells are statistically unstable.
+
+The size of a neighborhood is up to the analyst's discretion. Try to choose a neighborhood 
+size that reflects your understanding of biology and that makes sense for your biological question. There is a Goldilocks zone, however: very small neighborhoods produce sparse and noisy data, 
+and very large neighborhoods become inaccurate representations of a cell's 3D surroundings. 
+
+:::{.callout-note}
```

Comment:
comment: I changed the parenthetical to a call out note block. No action needed.